new package: goma

wortiz opened the pull request:
Adds a new package, Goma, finite element program

wortiz committed:
initial goma package

wortiz committed:
add version 7.0.0

alalazo commented on var/spack/repos/builtin/packages/goma/package.py:
```diff
@@ -0,0 +1,114 @@
+    version('main', branch='main')
+    version('develop', branch='develop')
+
+    variant('MDE', default=27, description="Set internal maximum DOF per element")
+    variant('MAX_PROB_VAR', default=15, description="Set internal maximum number of active equations")
+    variant('MAX_CONC', default=4, description="Set internal maximum number of species")
+    variant('MAX_EXTERNAL_FIELD', default=4, description="Set internal maximum number of external fields")
```
Two comments:
1. We usually have lowercase variants, separated by an underscore (since `-` can be a synonym to `~` in some contexts)
2. Can we define a set of allowed values for these 4 variants?

wortiz replied:
I think I've addressed this. There isn't any way to specify a range of numbers allowed is there? I did need to change them to strings to allow the value checker to work but cmake results seem fine.

alalazo replied:
Thanks!

alalazo commented on var/spack/repos/builtin/packages/goma/package.py:
```diff
@@ -0,0 +1,114 @@
+
+        args = []
+        args.extend([
+            '-DMDE=%s' % spec.variants['MDE'].value,
```
Here and below you can use `self.define` for having a terser recipe.

alalazo commented on var/spack/repos/builtin/packages/goma/package.py:
```diff
@@ -0,0 +1,114 @@
+            '-DTrilinos_DIR=%s' % spec['trilinos'].prefix
+        ])
+
+        if '+petsc' in spec:
+            args.extend([
+                '-DENABLE_PETSC=ON'
+            ])
+        else:
+            args.extend([
+                '-DENABLE_PETSC=OFF'
+            ])
```
Trying to write this by heart (so beware of typos):
```suggestion
        args.append(self.define_from_variant('ENABLE_PETSC', 'petsc'))
```
and so on for the variants below.

alalazo reviewed (changes requested):
I left a few comments. Let me know what you think.

wortiz committed:
Cleanup package with recommendations, add a couple variants

wortiz commented:
Thanks, I think I've addressed all the issues and cmake appears to pick up the correct libraries without specifying locations and just using `define_from_variant`.

wortiz committed:
match library order
